features/ALB002_Alignment-layout.feature

```diff
@@ -0,0 +1,50 @@
+@implementer-agreement
+@ALB
+Feature: ALB002 - Alignment Layout
+
+  Scenario: Agreement on nested attributes of IfcAlignment
+
+      Given A file with Schema Identifier "IFC4X3" or "IFC4X3_TC1" or "IFC4X3_ADD1"
+
+      Then Each IfcAlignment must be nested by exactly 1 instance(s) of IfcAlignmentHorizontal
+       And Each IfcAlignment must be nested by at most 1 instance(s) of IfcAlignmentVertical
+       And Each IfcAlignment must be nested by at most 1 instance(s) of IfcAlignmentCant  
+  
+    
+  Scenario: Agreement on attributes being nested within a decomposition relationship
+
+      Given a file with Schema Identifier "IFC4X3"
+
+      Then Each IfcAlignmentHorizontal must nest: only 1 instance(s) of IfcAlignment
+       And Each IfcAlignmentVertical must nest: only 1 instance(s) of IfcAlignment
+       And Each IfcAlignmentCant must nest: only 1 instance(s) of IfcAlignment
+  
+  Scenario: Agreement of structure of alignments segments
+
+      Given a file with Schema Identifier "IFC4X3"
+
+      Then Each IfcAlignmentHorizontal is nested by: a list of only instance(s) of IfcAlignmentSegment
+      Then Each IfcAlignmentVertical is nested by: a list of only instance(s) of IfcAlignmentSegment
+      Then Each IfcAlignmentCant is nested by: a list of only instance(s) of IfcAlignmentSegment
```

Comment:
Syntax question: when to use `:` in sentences, when not to?

Reply:
Ideally never, it is a temporary solution for a problem regarding placeholders in python. For example, the implementation for 'must nest only 1 instance' might in the future be re-used by 'must not be nested by at most 2', so {must: must not}, {nest: nested by}, {only: by only}.

Reply:
Ideally never, it is a temporary solution for a problem regarding placeholders for strings in python. There are a couple of solutions, but we still need to discuss a couple of theoretical issues regarding alignment before putting a lot of effort into this.

Reply:
Late reply, I solved it but the checks did not pass. That turned out to be due to an error with imports in the github environment.
The code now uses custom parsing of the gherkin sentence https://github.com/buildingSMART/ifc-gherkin-rules/pull/6/commits/4baf964890d619e7aaa8dcee74394152d1d7ac0c . That way, it works. 

The problem was in the sentence:
`Each Alignment is {relationship_type} {condition} of AlignmentVertical`
if you type 
`Each IfcAlignment is nested by a list of only instance(s) of IfcAlignmentVertical`
the behave module by default does not know we are searching for **'is nested by'** and **'a list of only'** and will return **'is**' and **'nested by a list of only',** which is not very helpful. 

With custom parsing (more precisely, with the pyparsing module) we can be very precise which terms to look for, which is good I think.